### PR TITLE
add put field endpoint

### PR DESCRIPTION
**Pull request by Tayef-Shah**
### Notion ticket link

[PUT field](https://www.notion.so/uwblueprintexecs/PUT-fields-id-endpoint-726ae92328b445c28da6b50385664939)

### Implementation description

- add PUT field to viewset
- add validator for field options
- update tests for view and serializer
- validation for changing the field order will come in another PR

### Steps to test

1. python manage.py test
2. load initial data and go to localhost:8000/fields/{id} and play around with the endpoint
* you can copy a field object JSON from a GET request to make testing easier

### What should reviewers focus on?

- Testing! Trying to make sure this endpoint works.

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by Tayef-Shah** [comment]
oh man, that commit message went horribly wrong lol

File: registration/tests/serializers/test_field.py [hunk 1]
```diff
@@ -91,23 +100,31 @@ def test_field_list_serializer(self):
...
-        field_request_invalid_order["order"] = 3
+        field_request_invalid_order["order"] = 1
         serializer = FieldSerializer(data=field_request_invalid_order, context=context)
+        self.assertFalse(serializer.is_valid())
+
+    def test_field_serializer__option_validator(self):
+        field_request_invalid_options = self.field_request
+        field_request_invalid_options["question_type"] = Field.SELECT
+
+        context = {"request": Request(APIRequestFactory().post("/fields/"))}
```
Comment (brittanylau) [inline registration/tests/serializers/test_field.py]: i think we can use `{ "request": None }` to be consistent with our other tests!

File: registration/tests/views/test_field.py [hunk 2]
```diff
@@ -57,18 +53,46 @@ def test_post_fields(self):
...
+                "question_type": Field.SELECT,
+                "is_default": False,
+                "options": ["Aunt, Uncle, Friend, Other"],
+                "order": Field.objects.filter(role=Field.GUEST).count() + 1,
+            },
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_method_not_allowed(self):
```
Comment (brittanylau) [inline registration/tests/views/test_field.py:86]: can we also verify that the get detail endpoint isn't allowed?

File: registration/validators.py [hunk 3]
```diff
@@ -79,3 +79,11 @@ def validate_field_order(field_order, role):
...
     ):
         raise ValidationError("Invalid order value")
+
+
+def validate_field_options(question_type, options):
+    Field = apps.get_model("registration", "Field")
+    if question_type in [Field.SELECT, Field.MULTIPLE_SELECT] and len(options) < 1:
+        raise ValidationError(
+            "Select and multi-select fields must have at least one option"
+        )
```
Comment (brittanylau) [inline registration/validators.py:89]: ah this is a good one, i didn't block users from doing this in the frontend - just put up https://github.com/uwblueprint/project-read-frontend/pull/113 so users can't submit invalid data

**Review by brittanylau — APPROVED** [review]
code lgtm, and tested with the frontend that things look good together 😄

File: registration/tests/serializers/test_field.py (hunk 1 above)
Comment (Tayef-Shah) [reply registration/tests/serializers/test_field.py]: changing it to a PUT request so it tests a smaller piece of code (passes through one less conditional)

**Pull request merged**